src/Libraries/CoreNodes/List.cs

```diff
@@ -52,6 +52,362 @@ public static bool ContainsItem(IList list, object item)
+        /// <search>union,setunion,set</search>
+        public static IList SetUnion(IList list1, IList list2)
+        {
+            foreach (var obj in list2)
+            {
+                if (!list1.Contains(obj))
+                {
+                    list1.Add(obj);
+                }
+            }
+            return list1;
+        }
+
+        /// <summary>
+        ///     Returns the index of the element in the given list.
+        /// </summary>
+        /// <param name="list">The list to find the element in.</param>
+        /// <param name="element">The element whose index is to be returned.</param>
+        /// <returns name="int">The index of the element in the list.</returns>
+        /// <search>index,indexof</search>
+        public static int IndexOf(IList list, object element)
+        {
+            return list.IndexOf(element);
+        }
+
+        /// <summary>
+        ///     Returns the number of false boolean values in the given list.
+        /// </summary>
+        /// <param name="list">The list find the false boolean values.</param>
+        /// <returns name="int">The number of false boolean values in the list.</returns>
+        /// <search>false,countfalse</search>
+        public static int CountFalse(IList list)
+        {
+            return CountBool(list, false);
+        }
+
+        /// <summary>
+        ///     Returns the number of true boolean values in the given list.
+        /// </summary>
+        /// <param name="list">The list find the true boolean values.</param>
+        /// <returns name="int">The number of true boolean values in the list.</returns>
+        /// <search>true,counttrue</search>
+        public static int CountTrue(IList list)
+        {
+            return CountBool(list, true);
+        }
+
+        /// <summary>
+        ///     Returns the number of the specified boolean values in the given list.
+        /// </summary>
+        /// <param name="list">The list find the true boolean values.</param>
+        /// <returns name="int">The number of the specified boolean value in the list.</returns>
+        private static int CountBool(IList list, bool b)
+        {
+            int count = 0;
+            foreach (var obj in list)
+            {
+                if ((obj is bool) && ((bool)obj == b)) count++;
+                else if ((obj is IList))
+                {
+                    count += CountBool((IList)obj, b);
+                }
+            }
+            return count;
+        }
+
+        /// <summary>
+        ///     Returns the flattened 1D list of the multi-dimensional input list.
+        /// </summary>
+        /// <param name="list">The list to be flattened.</param>
+        /// <returns name="list">The flattened 1D list.</returns>
+        /// <search>flatten,1D</search>
+        public static IList Flatten(IList list)
+        {
+            return Flatten(list, GetDepth(list), new List<object>());
```

Comment:
I initially mapped builtin Flatten to List.Flatten in Thunderstorm.  Unfortunately, this results in a change in output for the case where builtin Flatten is passed a simple value rather than a list.  It currently returns the value unmodified rather than putting it in a list.